### ISSUE TITLE
Develop proper formula macro for GAM specifications

### ISSUE THREAD

**Issue by hendersontrent** [hendersontrent/GAM.jl#46]
Currently, a string is used, but a proper formula macro would be desirable.

**Comment by behinger** [comment]
I implemented a simple GAM-style formula syntax for BSplines and CircularBSplines for Unfold.jl (using BSplineKit) here: https://github.com/unfoldtoolbox/Unfold.jl/blob/main/ext/UnfoldBSplineKitExt/splinepredictors.jl

Maybe this is a useful starting point for this implementation

**Comment by iabraham** [comment]
Could an option be to borrow the "Formula language" from [`StatsModels`](https://juliastats.org/StatsModels.jl/stable/formula/) ?

**Comment by rsenne** [comment]
Any idea on how we want to tackle this? I have started working on a branch for this but have yet committed to a design philosophy. Happy to drop a few ideas if desired

**Comment by behinger** [comment]
I agree with iabraham, StatsModels is the way to go in terms of Formula-Syntax. I think it should offer everything one needs
